### PR TITLE
add Trusted Language Extensions

### PR DESCRIPTION
**Pull request by michelp**
## What kind of change does this PR introduce?

Adds Trusted Language Extensions (TLE) Extension

### CODE REVIEW

**Review by kiwicopple — APPROVED** [review]

**Comment by michelp** [comment]
There's a `pgtle_admin` role that is necessary to install/update/uninstall TLEs that should be conveyed to the `postgres` role at some point in the setup flow, not 100% positive where that should go someone on @supabase/backend have any suggestions?

**Comment by darora** [comment]
Initial instinct would be to suggest `migrations/db/init-scripts/00000000000000-initial-schema.sql` unless this role is categorically different from the ones we currently deal with there?

**Comment by michelp** [comment]
The role is created by the extension at CREATE EXTENSION time, so I don't think anything special needs to be done?  As postgres has CREATEROLE already it should be able to self-grant it, maybe this requires a tweak to supautils?

https://github.com/aws/pg_tle/blob/c12bfc37fb5c1066339aa66c5bad9d04baebe96c/pg_tle--1.0.0.sql#L283

**Review by darora — APPROVED** [review]
@sweatybridge fyi doesn't add the extension to the separate docker build

**Comment by dragarcia** [comment]
* Attempt PR for installing TLE on Docker here: https://github.com/supabase/postgres/pull/542

**Comment by soedirgo** [comment]
Re: `pgtle_admin`, we can handle it the same way we handle [`pgsodium_*` roles](https://github.com/supabase/postgres/blob/47c60bf1fb45440d7ab4bcd7405770c39bdffb00/migrations/db/migrations/20221207154255_create_pgsodium_and_vault.sql).

File: ansible/Dockerfile [hunk 1]
```diff
@@ -681,6 +682,27 @@ RUN --mount=type=cache,source=/ccache,target=/ccache,from=public.ecr.aws/t3w2s2c
...
+ARG pg_tle_release
+ARG pg_tle_release_checksum
+ADD --checksum=${pg_tle_release_checksum} \
+    "https://github.com/aws/pg_tle/archive/refs/tags/v${pg_tle_release}.tar.gz" \
+    /tmp/pg_tle.tar.gz
+RUN tar -xvf /tmp/pg_tle.tar.gz -C /tmp && \
+    rm -rf /tmp/pg_tle.tar.gz
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    flex \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (dragarcia) [inline ansible/Dockerfile]:
```suggestion
```

**Pull request merged**